### PR TITLE
Upgrade eslint and eslint config

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
- Upgrades to latest [eslint](https://www.npmjs.com/package/eslint) and  [eslint-config-eq-author](https://www.npmjs.com/package/eslint-config-eq-author) and various other dependant packages

### How to review 
- Ensure tests pass
- Ensure [new linting rules](https://github.com/ONSdigital/eslint-config-eq-author/pull/7) work as expected

### CODE REVIEW

**Comment by adammahmood** [comment]
[Add ESLint rule: https://eslint.org/docs/rules/no-unused-vars](https://trello.com/c/gF1Ybe0D/641-add-eslint-rule-https-eslintorg-docs-rules-no-unused-vars)

**Comment by adammahmood** [comment]
[Add a lint rule preventing it.only from being commited](https://trello.com/c/oBf9S4Me/642-add-a-lint-rule-preventing-itonly-from-being-commited)

File: src/components/Loading/index.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const colorChange = keyframes`
   }
 `;
 
-const Container = styled(({ height, ...props }) => <div {...props} />)`
+const Container = styled(({ ...props }) => <div {...props} />)`
```
Comment (tgandrews) [inline src/components/Loading/index.js]: Is this now just `styled.div`?

Comment (adammahmood) [reply src/components/Loading/index.js]: Yep.  Fixed.

File: package.json [hunk 2]
```diff
@@ -146,6 +143,7 @@
     "react-hot-loader": "^4.3.4",
     "react-inlinesvg": "^0.7.5",
     "react-modal": "^3.4.4",
+    "react-reconciler": "^0.15.0",
```
Comment (adammahmood) [inline package.json]: @SamGodwin2 This is required here: https://github.com/ONSdigital/eq-author/blob/b0f76d8863a88ea8a24e936f9f650c1340dace6d/src/tests/setup/ReactSixteenAdapter.js#L25

Comment (adammahmood) [reply package.json]: The `ReactSixteenAdapter.js` needs to be removed at some point in the future.  It's was a temporary workaround for testing React Context with Enzyme until they supported it.

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "test:storybook": "jest --testRegex ^.+\\.storytest\\.js$",
     "test:e2e": "./scripts/run_e2e_tests_author.sh",
     "lint": "yarn lint:js && yarn lint:css",
-    "lint:js": "eslint src --ext .graphql --ext .js --max-warnings=0",
+    "lint:js": "eslint src cypress --ext .graphql --ext .js --max-warnings=0",
```
Comment (tgandrews) [inline package.json]: Could we just look at root and use an `.eslintignore` file to ignore particular folders. But maybe not in this PR.

**Pull request merged**